src/api/app/controllers/webui/request_controller.rb

```diff
@@ -402,6 +384,25 @@ def accept_request
     end
   end
 
+  def prepare_bsrequest(action, params)
+    req = BsRequest.new(state: 'new', description: params[:description])
+    params.delete :description
+    klass_to_call = case action
```

Comment:
is this case really needed? why not giving the method the class name? or calculate the class name just with `"bs_request#{action}".classify"`... but the class name is not that complicate that we can not write it in the mehod itself instead of a case here... :confounded:

Reply:
it's possible, but would I prefer to whitelist the classes "allowed" to avoid someone abusing this construction to reach others classes and maybe leveraging some security breach. Any suggestion?

Reply:
I am fine with the case, but I would also move it to it's own method. And then you could do something like `action_class.new(params) ` instead of using send

Reply:
security? the action to call is in our code.

I don't see benefit in this. I would just use the class in every method. We are making this much complicate to follow with no reason. :confused:

Reply:
complicated to follow? can you please provide me a code suggestion?